src/main/cpp/bgjs/modules/BGJSGLModule.cpp

```diff
@@ -954,11 +979,13 @@ void BGJSGLModule::js_canvas_constructor(const v8::FunctionCallbackInfo<v8::Valu
 	}
 	Local<Object> obj = args[0]->ToObject();
 	BGJSCanvasGL* canvas = new BGJSCanvasGL();
-	Local<Value> external = obj->GetInternalField(0);
-	canvas->_view = externalToClassPtr<BGJSGLView>(external);
+	canvas->_view = JNIV8Wrapper::wrapObject<BGJSGLView>(args[0]->ToObject());
 
 	Local<Object> fn = Local<Function>::New(isolate, BGJSGLModule::g_classRefCanvasGL)->NewInstance();
 	fn->SetInternalField(0, External::New(isolate, canvas));
+    CanvasCallbackHolder* persistentHolder = new CanvasCallbackHolder();
+    persistentHolder->persistent.Reset(isolate, fn);
+    persistentHolder->persistent.SetWeak((void*)canvas, js_canvas_destruct, WeakCallbackType::kParameter);
```

Comment:
If you only forward the canvas, and delete it in the destruct callback, the CanvasCallbackHolder instance will leak.
=> use persistentHolder as parameter for SetWeak, add the canvas object as a member on CanvasCallbackHolder;
in js_canvas_destruct delete both the member AND the holder forwarded data parameter